Add KEV flag to ticket documents

felddy opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

Add the new KEV flag to the details section of ticket documents.  
Generate notification documents for KEV flags as we do for high and criticals. 

## 💭 Motivation and context ##

Resolves cisagov/cyhy-system#41 and resolves cisagov/cyhy-system#43
Note: Sorted a bunch of thangs.  You can probably just focus on reviewing the non-sorting commits. 

## 🧪 Testing ##

- [x] Tested in development cyhy environments by adding a host that had a CVE listed in the KEVs.  Notification documents were generated as expected.

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.

felddy committed:
Sort dictionary keys and other assignments

felddy committed:
Add KEV flag to ticket docment details

felddy committed:
Add notification when KEV is true for new tickets

Co-authored-by: dav3r <dav3r@users.noreply.github.com>

dav3r reviewed (approved):
KEVtastic! 💎

mcdonnnj commented on cyhy/db/ticket_manager.py:
```diff
@@ -77,44 +77,50 @@ def __check_false_positive_expiration(self, ticket, time):
 
     def __generate_ticket_details(self, vuln, ticket, check_for_changes=True):
-        """generates the contents of the ticket's details field using NVD data.
+        """Generates the contents of the ticket's details field using NVD data.
         if check_for_changes is True, it will detect changes in the details,
```
Since we're already editing this docstring the summary should be in the imperative mood and there should be a blank line after the summary.

```suggestion
        """Generate the contents of the ticket's details field using NVD data.

        If check_for_changes is True, it will detect changes in the details,
```

mcdonnnj commented on cyhy/db/ticket_manager.py:
```diff
@@ -244,8 +250,8 @@ def open_ticket(self, vuln, reason):
 
-        # Create notifications for Highs (3) or Criticals (4)
-        if new_ticket["details"]["severity"] > 2:
+        # Create notifications for Highs (3) or Criticals (4) or if KEV is true
```
Just something to break up the chain of `or`s since the first two are part of one logical check.

```suggestion
        # Create notifications for Highs (3) or Criticals (4), or if KEV is true
```

mcdonnnj commented on cyhy/db/ticket_manager.py:
```diff
@@ -577,7 +583,7 @@ def clear_vuln_latest_flags(self):
 
 class IPTicketManager(object):
-    """Handles the closing of tickets for a host scan (NETSCAN) """
+    """Handles the closing of tickets for a host scan (NETSCAN)"""
```
```suggestion
    """Handle the closing of tickets for a host scan (NETSCAN)."""
```

mcdonnnj reviewed (changes requested):
Makes sense. I had some suggestions for your consideration.

felddy committed:
Apply suggestions from code review

- Since we're already editing this docstring the summary should be in the imperative mood and there should be a blank line after the summary.
- Add comma: Just something to break up the chain of ors since the first two are part of one logical check.
- Correct tense of method comment.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>
